modules/ts/src/ts_gtest.cpp

```diff
@@ -1067,6 +1067,7 @@ class GTEST_API_ UnitTestImpl {
   void AddTestInfo(Test::SetUpTestCaseFunc set_up_tc,
                    Test::TearDownTestCaseFunc tear_down_tc,
                    TestInfo* test_info) {
+#ifndef OPENCV_DISABLE_THREAD_SUPPORT
     // In order to support thread-safe death tests, we need to
     // remember the original working directory when the test program
```

Comment:
How is directory related to disabled threads?
Suggesting to revert this change.

Reply:
Right - this is something that should probably be guarded by the `OPENCV_HAVE_FILESYSTEM_SUPPORT`. I have removed it from this PR, it might show up with a new guard in #19921